k8s/cursor-initializator-clickhouse/cursor-initializator-clickhouse.py

```diff
@@ -33,104 +22,31 @@ def get_logger():
+    logger.info("Looking for nodes matching %s", args.node_filter)
+
+    client = clickhouse_connect.get_client(interface="https", dsn=args.url, password=args.password)
+
+    result = client.query(
+        f"""
+        SELECT
+            `ic_node_id`,
+            `cursor`
+        FROM
+            `ic_boundary_cursor_distributed` FINAL
+        WHERE
+            match(`ic_node_id`, '{args.node_filter}')
         """
     )
 
-    logger.info("Found %s tables", len(tables))
-    if not all([table in tables for table in args.tables]):
-        logger.error("Table %s not found", args.table)
-        sys.exit(1)
-
-    logger.info("Table found")
-
-    aggregated = {}
-
-    for table in args.tables:
-        logger.info("Looking for nodes in table %s", table)
-
-        field = "_HOSTNAME"
-        if table == "ic":
-            field = "ic_node"
-
-        command = get_last_cursor_for_node(table, args.node_filter, field)
-        logger.info("Executing command: \n%s", command)
-        response = client.command(command)
-
-        if not isinstance(response, list):
-            # should happen only if the result is empty
-            response = []
-
-        mapped = [item for line in response for item in line.split("\n")]
-
-        for i in range(0, len(mapped), 4):
-            node = mapped[i]
-            timestamp = mapped[i + 1]
-            cursor = mapped[i + 2]
-            job = mapped[i + 3]
-
-            if node not in aggregated:
-                aggregated[node] = {}
-            aggregated[node][job] = {
-                "cursor": cursor,
-                "timestamp": timestamp,
-            }
-
-    logger.info("Dumping aggregated cursors: \n%s", json.dumps(aggregated, indent=2, sort_keys=True))
-    created = 0
-    for node in aggregated:
-        for job in aggregated[node]:
-            file_name = node
-            if len(node.split("-")) == 2:
-                if job == "host_node_exporter":
-                    file_name = f"{file_name}-host"
-                elif job == "node_exporter":
-                    file_name = f"{file_name}-guest"
-
-            path = os.path.join(args.output_dir, f"{file_name}-{job}-source")
-            if not os.path.exists(path):
-                os.mkdir(path)
-            else:
-                logger.warning("Directory already exists, maybe this shouldn't be overriden? %s", path)
-
-            checkpointer = os.path.join(path, "checkpoint.txt")
-            with open(checkpointer, "w", encoding="utf-8") as f:
-                f.write(aggregated[node][job]["cursor"] + "\n")
-                created += 1
+    for r in result.result_rows:
+        with open(os.path.join(args.output_dir, f"{r[0]}.txt"), "w", encoding="utf-8") as f:
+            f.write(f"{r[1]}\n")
 
-    logger.info("Successfully initialized cursors %s on path %s", created, args.output_dir)
+    logger.info(f"Successfully initialized {len(result.result_rows)} cursors on path {args.output_dir}")
```

Comment:
Lazy log formatting is the recommended way to go: https://stackoverflow.com/questions/4148790/lazy-logger-message-string-evaluation

Reply:
Ok, not that performance matters here :)